cli/tests/default/unit/test_engine_type.py

```diff
@@ -2,79 +2,127 @@
 
 from semgrep.app.scans import ScanHandler
 from semgrep.engine import EngineType as ET
+from semgrep.error import SemgrepError
 from semgrep.meta import GitMeta
 
 
 @pytest.mark.quick
+@pytest.mark.parametrize("is_supply_chain_only", [True, False])
+@pytest.mark.parametrize("is_secrets_scan", [True, False])
 @pytest.mark.parametrize(
-    ("is_cloud_flag_on", "is_ci_scan_full", "enable_pro_diff", "requested", "expected"),
+    "engine_flag", [None, ET.OSS, ET.PRO_LANG, ET.PRO_INTRAFILE, ET.PRO_INTERFILE]
+)
+@pytest.mark.parametrize(
+    (
+        "logged_in",
+        "is_interfile_flag_on",
+        "is_ci_scan_full",
+        "interfile_diff_scan_enabled",
+        "expected_default",
+    ),
     [
+        # These combinations define the expected engine for Code scans when no
+        # engine is explicitly requested via a CLI argument. Requesting the engine
+        # in the CLI or running Secrets or Supply Chain may change the engine used.
+        #
+        # We assume some invariants and thus don't test them:
+        # - not logged_in -> is_interfile_flag_on is None, is_ci_scan_full is None
+        # - is_interfile_flag_on is None -> is_ci_scan_full is None
+        #
         # semgrep scan
-        (False, None, False, None, ET.OSS),
-        (False, None, False, ET.OSS, ET.OSS),
-        (False, None, False, ET.PRO_LANG, ET.PRO_LANG),
-        (False, None, False, ET.PRO_INTRAFILE, ET.PRO_INTRAFILE),
-        (False, None, False, ET.PRO_INTERFILE, ET.PRO_INTERFILE),
+        (False, None, None, False, ET.OSS),
+        (False, None, None, True, ET.OSS),
+        (True, None, None, False, ET.OSS),
+        (True, None, None, True, ET.OSS),
+        # semgrep ci, not logged in
```

Comment:
I thought one needed to be logged in to even do anything with `semgrep ci`, since the rules etc need to come from the App.

Reply:
Nope! From `ci.py`:
```
    if not token and not config:
        # Not logged in and no explicit config
        logger.info(
            "run `semgrep login` before using `semgrep ci` or use `semgrep scan` and set `--config`"
        )
        sys.exit(INVALID_API_KEY_EXIT_CODE)
    elif not token and config:
        # Not logged in but has explicit config
        pass
```

The second condition explicitly allows `semgrep ci` with no token but `--config`. (Note that if you do pass a token, you aren't allowed to give it `--config`, and we ask them to log out instead. I suspect we are going to want to change this behavior but that's very out of scope.)